tests/test_gpu_openmp.py

```diff
@@ -194,7 +209,29 @@ def test_timeparallel_reduction(self):
         assert not tree.root.pragmas
         assert len(tree[1].pragmas) == 1
         assert tree[1].pragmas[0].value ==\
-            'omp target teams distribute parallel for collapse(3) reduction(+:f[0])'
+            ('omp target teams distribute parallel for device(devicenum) collapse(3)'
+             ' reduction(+:f[0])')
+
+    @pytest.mark.parallel(mode=1)
+    @switchconfig(platform='nvidiaX')
+    def test_gpu_direct(self):
+        grid = Grid(shape=(3, 3, 3))
+
+        u = TimeFunction(name='u', grid=grid)
+
+        op = Operator(Eq(u.forward, u.dx+1), opt=('advanced', {'gpu-direct': True}))
+
+        assert str(op.body[0].body[0]) == 'int rank = 0;'
```

Comment:
this should be a separate test -- test_init_omp_env or something along these lines

Reply:
and it should be one of the first tests of the class